### PR TITLE
mdbook-footnote: init 0.1.1

### PR DESCRIPTION
**Pull request by brianmcgillion**
## Description of changes

Added new package mdbook-footnote version 0.1.1. mdbook-footnote is a preprocessor for mdbook to support the inclusion of automatically numbered footnotes.

https://github.com/daviddrysdale/mdbook-footnote

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/mdbook-footnote/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ lib, stdenv, fetchFromGitHub, rustPlatform, CoreServices }:
```
Comment (MikaelFangel) [inline pkgs/tools/text/mdbook-footnote/default.nix:1]:
```suggestion
{ lib
, stdenv
, fetchFromGitHub
, rustPlatform
, CoreServices 
}:
```

File: pkgs/tools/text/mdbook-footnote/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+rustPlatform.buildRustPackage rec {
+  pname = "mdbook-footnote";
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner = "daviddrysdale";
+    repo = pname;
```
Comment (MikaelFangel) [inline pkgs/tools/text/mdbook-footnote/default.nix:8]:
```suggestion
    repo = "mdbook-footnote";
```

File: pkgs/tools/text/mdbook-footnote/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    rev = version;
+    sha256 = "sha256-WUMgm1hwsU9BeheLfb8Di0AfvVQ6j92kXxH2SyG3ses=";
```
Comment (MikaelFangel) [inline pkgs/tools/text/mdbook-footnote/default.nix:10]:
```suggestion
    hash = "sha256-WUMgm1hwsU9BeheLfb8Di0AfvVQ6j92kXxH2SyG3ses=";
```

File: pkgs/tools/text/mdbook-footnote/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  };
+
+  cargoSha256 = "sha256-Ig+uVCO5oHIkkvFsKiBiUFzjUgH/Pydn4MVJHb2wKGc=";
+
+  buildInputs = lib.optionals stdenv.isDarwin [ CoreServices ];
+
+  meta = with lib; {
+    description =
+      "A preprocessor for mdbook to support the inclusion of automatically numbered footnotes.";
```
Comment (MikaelFangel) [inline pkgs/tools/text/mdbook-footnote/default.nix:19]:
```suggestion
      "A preprocessor for mdbook to support the inclusion of automatically numbered footnotes";
```

File: pkgs/tools/text/mdbook-footnote/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    homepage = "https://github.com/daviddrysdale/mdbook-footnote";
+    license = [ licenses.asl20 ];
```
Comment (MikaelFangel) [inline pkgs/tools/text/mdbook-footnote/default.nix:21]:
```suggestion
    license = licenses.asl20;
```

File: pkgs/tools/text/mdbook-footnote/default.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    maintainers = with maintainers; [ brianmcgillion ];
```
Comment (MikaelFangel) [inline pkgs/tools/text/mdbook-footnote/default.nix:22]: If the program results in a binary that is expected to be run you should include the meta tag `mainProgram`.

**Review by MikaelFangel — COMMENTED** [review]
Thank you for your contribution. By my quick skim of your derivation I found some improvements that could be made so it's more idiomatic and correct in terms of nixpkgs (:

**Comment by h7x4** [comment]
Please lowercase the commit message for the maintainer list commit, and refer to the attribute name rather than the full name

```
maintainers: add brianmcgillion
```

File: pkgs/tools/text/mdbook-footnote/default.nix (hunk 6 above)
Comment (brianmcgillion) [reply pkgs/tools/text/mdbook-footnote/default.nix:26]: It is a preprocessor plugin, that is registered in the `book.toml` and called during a run of `mdbook`, for your documentation project

**Comment by brianmcgillion** [comment]
@MikaelFangel, @h7x4 Thank you both for your quick response and explanations!

**Review by zimbatm — APPROVED** [review]

File: pkgs/tools/text/mdbook-footnote/default.nix [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+, CoreServices
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "mdbook-footnote";
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner = "daviddrysdale";
+    repo = "mdbook-footnote";
+    rev = version;
```
Comment (zimbatm) [inline pkgs/tools/text/mdbook-footnote/default.nix:14]:
```suggestion
    rev = "v${version}";
```
The tag has a "v" prefix.

**Comment by zimbatm** [comment]
Result of `nixpkgs-review pr 268866` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>mdbook-footnote</li>
  </ul>
</details>

**Pull request merged**